### PR TITLE
Removing framework related and unused files.

### PR DESCRIPTION
**Pull request by quetzal42**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

Description
Provide the information we need to review your PR. What problem does the pull request solve? "Bug fix" is not a good description.

Related Issue
If you opened an issue before creating the PR, point to it here.

Context
What do we need to know about your development environment, tools, target, and so on. Screenshots are always helpful if there is a UI element to this PR.

### CODE REVIEW

File: docs/plantuml/diagrams/3DMagneticCore.png
Comment (djaumann) [inline docs/plantuml/diagrams/3DMagneticCore.png:1]: Didn't we say that we don't want to publish the Plant UML files? Or do I remember it wrong?

File: docs/plantuml/diagrams/3DMagneticCore.puml
Comment (djaumann) [inline docs/plantuml/diagrams/3DMagneticCore.puml:1]: Same here

**Review by djaumann — APPROVED** [review]

**Review by jaenrig-ifx — APPROVED** [review]
From this I cannot judge how this repository is consumed by the framework repository ones. But removing all the framework relevant sources and moving them to their respective repos seem fine :)

File: docs/plantuml/diagrams/3DMagneticCore.png
Comment (quetzal42) [reply docs/plantuml/diagrams/3DMagneticCore.png:1]: I believe an architectural overview should be given and also included in the docs later. In order to be in sync with the code I would therefore deliver the png for easy viewing and the source below as well, but not the many other PLANTUML files previously in the release.

File: docs/plantuml/diagrams/3DMagneticCore.puml
Comment (quetzal42) [reply docs/plantuml/diagrams/3DMagneticCore.puml:1]: See above

File: docs/plantuml/diagrams/3DMagneticCore.png
Comment (djaumann) [reply docs/plantuml/diagrams/3DMagneticCore.png:1]: Sounds reasonable :)

**Review by ederjc — APPROVED** [review]

**Pull request merged**